### PR TITLE
UAF-3636 UAF-3389 BUG - Cannot Create Divisions for / Unable to Edit Foreign Vendors

### PR DESCRIPTION
**Pull request by ghost**
Fixes to validation of vendor tax number/vendor tax type if Vendor or Vendor parent has no tax number (foreign vendor). This bug fix is for both UAF-3636 and UAF-3389.

New files:
1. edu.arizona.kfs.vnd.businessobject.options.TaxPayerTypeWithNoneValuesFinder.java
2. edu.arizona.kfs.vnd.businessobject.datadictionary.VendorHeader.xml

Files changed:
1. edu.arizona.kfs.vnd.document.validation.impl.VendorRule.java
2. org.kuali.kfs.vnd.document.validation.impl.VendorRule.java

### CODE REVIEW

File: kfs-core/src/main/java/edu/arizona/kfs/vnd/document/validation/impl/VendorRule.java [hunk 1]
```diff
@@ -64,4 +68,43 @@ private boolean processSupplierDiversityValidation(MaintenanceDocument document)
...
+    @Override
+    protected boolean validateTaxTypeAndTaxNumberBlankness(VendorDetail vendorDetail) {
+        boolean valid = true;
+        boolean isParent = vendorDetail.isVendorParentIndicator();
+        String vendorTaxTypeCode = vendorDetail.getVendorHeader().getVendorTaxTypeCode();
+        if (!StringUtils.isBlank(vendorDetail.getVendorHeader().getVendorTaxNumber()) && VendorConstants.NONE.equals(vendorTaxTypeCode)) {
+        	if (isParent) {
+                putFieldError(VendorPropertyConstants.VENDOR_TAX_TYPE_CODE, VendorKeyConstants.ERROR_VENDOR_TAX_TYPE_CANNOT_BE_BLANK);
+            }
+            valid &= false;
```
Comment (nataliac06) [inline kfs-core/src/main/java/edu/arizona/kfs/vnd/document/validation/impl/VendorRule.java:93]: No need for the &= operator here, valid = false should do it.

File: kfs-core/src/main/java/edu/arizona/kfs/vnd/document/validation/impl/VendorRule.java [hunk 2]
```diff
@@ -64,4 +68,43 @@ private boolean processSupplierDiversityValidation(MaintenanceDocument document)
...
+        }
+        else if (StringUtils.isBlank(vendorDetail.getVendorHeader().getVendorTaxNumber()) && !VendorConstants.NONE.equals(vendorTaxTypeCode)) {
+        	if (isParent) {
+                putFieldError(VendorPropertyConstants.VENDOR_TAX_TYPE_CODE, VendorKeyConstants.ERROR_VENDOR_TAX_TYPE_CANNOT_BE_SET);
+            }
+            valid &= false;
```
Comment (nataliac06) [inline kfs-core/src/main/java/edu/arizona/kfs/vnd/document/validation/impl/VendorRule.java:99]: Same as above.

File: kfs-core/src/main/java/edu/arizona/kfs/vnd/document/validation/impl/VendorRule.java (hunk 1 above)
Comment (ghost) [reply kfs-core/src/main/java/edu/arizona/kfs/vnd/document/validation/impl/VendorRule.java:93]: Discussed with Natalia and confirmed coded like this in delivered code. Agreed not worth another commit to fix this.

File: kfs-core/src/main/java/edu/arizona/kfs/vnd/document/validation/impl/VendorRule.java (hunk 2 above)
Comment (ghost) [reply kfs-core/src/main/java/edu/arizona/kfs/vnd/document/validation/impl/VendorRule.java:99]: Discussed with Natalia and confirmed coded like this in delivered code. Agreed not worth another commit to fix this.

**Pull request merged**